sklearn/preprocessing/data.py

```diff
@@ -1674,22 +1674,22 @@ def _transform_selected(X, transform, selected="all", copy=True):
         return transform(X)
     else:
         X_sel = transform(X[:, ind[sel]])
-        X_not_sel = X[:, ind[not_sel]]
+        X_not_sel = X[:, ind[not_sel]].astype(dtype)
```

Comment:
nitpick: `ind` seems not needed here. 

`X[:, sel]` and `X[:, ~sel]` should work as fine.